### PR TITLE
Article 065: Information hiding

### PR DESCRIPTION
**Pull request by Garcko**
#### Length of text ####
 :page_with_curl: short

####  Difficulty to read ####
 :mouse:  light

####  Difficulty to understand #### 
 :school_satchel:  easy 

####  Type of text #### 

 :bowtie:  informative 

####  Type of header #### 
 

 :boom:  catchy

### CODE REVIEW

File: articles/article-065-Information hiding.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+ 
+## Figure
+* A vending machine from the side, where we can see the interface(the button and a coin slit, and then the complexity of the inside)
+
+
+
+## Optional text
+<!-- BEGIN OPTIONAL -->
+* Here only the interface information hiding is mentioned and not  on the  packages  level( more user to program and not program to program).
```
Comment (s3lph) [inline articles/article-065-Information hiding.md]: Please don't put review notes into the optional text section. Instead, add them as a comment to the pull request.

**Comment by Teifun2** [comment]
**Plenum:**
Problems:
- Title is not connected to the text (Maybe explain why your hiding)

Idea:
- Its about service.
- Hiding to not be so vulnerable
- Discussion about example.
- Graphics 3xVending Machine, with person, gear, and electronics.

**Comment by MoiraZuber** [comment]
Added an image suggestion. I had trouble realizing the suggestion with the three vending machines, so I only did one and kept it rather abstract.

**Comment by Garcko** [comment]
Changed Title, and added a sentence to the vulnerability part, 
For the image I really like it 👍

**Comment by Fjolyo** [comment]
We think the content of this article is good, but the sentences sound very German and are often very long.

Example:
>The idea for information hiding is to reduce complexity and to interact not with the information/data directly, but only indirectly via an interface.

Could be written as:
The idea of *information hiding* is to reduce complexity and to not interact with the data directly. Instead, an *interface* is used.

Maybe the grammar team can help here?

**Comment by noverney** [comment]
I reworked a few of the sentences and just have the entire new version below:

The idea for information hiding is to reduce complexity and indirectly interact with the information/data using an interface. But why would we do this?

Suppose we want to get a drink from a vending machine. We only need to select the drink we want and then pay for it. We never need to see the inner workings or the logic behind the vending machine.

This is the same for computer program. We use interfaces to hide the information/methods.
This also means that we can interchange the hidden information easily as long as we
do not change the interface. Furthermore, since we do not know exactly how the programs or machines work, we will have a harder time to manipulate them.

**Comment by Garcko** [comment]
I changed to the new version below, sorry for the long waiting

**Comment by Teifun2** [comment]
Looks god to me

**Pull request merged**